src/Storage/Controllers/InstancesController.cs

```diff
@@ -183,7 +184,30 @@ public class InstancesController : ControllerBase
                     return BadRequest("Invalid InstanceOwnerIdentifier.");
                 }
 
-                (string person, string orgNo) = InstanceHelper.SeparatePersonAndOrgNo(instanceOwnerIdType, instanceOwnerIdValue);
+                string person = null;
+                string orgNo = null;
+
+                if (Enum.TryParse<PartyType>(instanceOwnerIdType, true, out PartyType partyType))
+                {
+                    if (partyType == PartyType.Person)
+                    {
+                        if (instanceOwnerIdValue.Length != 11)
+                        {
+                            return BadRequest("Person number needs to be exactly 11 digits.");
+                        }
+
+                        person = instanceOwnerIdValue;
+                    }
+                    else if (partyType == PartyType.Organisation)
+                    {
+                        if (instanceOwnerIdValue.Length != 9)
```

Comment:
Same as for Person it's now possible to get past the check with an invalid organization number by including letters.